sdv/constraints/tabular.py

```diff
@@ -484,6 +485,14 @@ def is_valid(self, table_data):
             low = cast_to_datetime64(low, self._low_datetime_format)
             high = cast_to_datetime64(high, self._high_datetime_format)
 
+            if self._low_datetime_format != self._high_datetime_format:
+                low, high = match_datetime_precision(
```

Comment:
Idk if this is a costly operation but I think we only needed to do this when `strict_boundaries` is False